File: rezmateportal_app/lib/core/constants/availability_status.dart
/// حالات الإتاحة
/// Availability Status constants
class AvailabilityStatus {
  static const String available = 'Available';
  static const String booked = 'Booked';
  static const String blocked = 'Blocked';
  static const String maintenance = 'Maintenance';
  static const String ownerUse = 'OwnerUse';

  /// الأسماء بالعربية
  static const Map<String, String> arabicNames = {
    available: 'متاح',
    booked: 'محجوز',
    blocked: 'محظور',
    maintenance: 'صيانة',
    ownerUse: 'استخدام المالك',
  };

  /// الألوان المرتبطة بكل حالة
  static const Map<String, int> statusColors = {
    available: 0xFF4CAF50, // Green
    booked: 0xFFFF9800, // Orange
    blocked: 0xFFF44336, // Red
    maintenance: 0xFF9E9E9E, // Grey
    ownerUse: 0xFF2196F3, // Blue
  };

  /// التحقق من صحة الحالة
  static bool isValidStatus(String status) {
    return status == available ||
        status == booked ||
        status == blocked ||
        status == maintenance ||
        status == ownerUse;
  }

  /// الحصول على الاسم بالعربية
  static String getArabicName(String status) {
    return arabicNames[status] ?? status;
  }

  /// الحصول على لون الحالة
  static int getStatusColor(String status) {
    return statusColors[status] ?? 0xFF9E9E9E;
  }

  /// الحصول على جميع الحالات
  static List<String> getAllStatuses() {
    return [available, booked, blocked, maintenance, ownerUse];
  }
}

File: rezmate_backend/YemenBooking.Core/Enums/AvailabilityStatus.cs
namespace YemenBooking.Core.Enums;

/// <summary>
/// تعداد حالات الإتاحة
/// Availability Status enumeration
/// </summary>
public static class AvailabilityStatus
{
    /// <summary>
    /// متاح
    /// Available
    /// </summary>
    public const string Available = "Available";
    
    /// <summary>
    /// محجوز
    /// Booked
    /// </summary>
    public const string Booked = "Booked";
    
    /// <summary>
    /// محظور
    /// Blocked
    /// </summary>
    public const string Blocked = "Blocked";
    
    /// <summary>
    /// صيانة
    /// Maintenance
    /// </summary>
    public const string Maintenance = "Maintenance";
    
    /// <summary>
    /// استخدام المالك
    /// Owner Use
    /// </summary>
    public const string OwnerUse = "OwnerUse";
    
    /// <summary>
    /// التحقق من صحة الحالة
    /// Validate status value
    /// </summary>
    public static bool IsValidStatus(string status)
    {
        return status == Available || 
               status == Booked || 
               status == Blocked || 
               status == Maintenance || 
               status == OwnerUse;
    }
    
    /// <summary>
    /// الحصول على جميع الحالات
    /// Get all statuses
    /// </summary>
    public static string[] GetAllStatuses()
    {
        return new[] { Available, Booked, Blocked, Maintenance, OwnerUse };
    }
}
